### PR TITLE
Convert video settings from xblock metadata into the redux store. In-progress draft. TNL-10009.

### PR DESCRIPTION
**Pull request by kenclary**
https://2u-internal.atlassian.net/browse/TNL-10009

This change loads data from blockValue to the video redux store.

### CODE REVIEW

**Comment by KristinAoki** [comment]
Have your rebased with `main` and checked that this works correctly with the updated widgets?

File: src/editors/data/redux/thunkActions/video.js [hunk 1]
```diff
@@ -1,9 +1,127 @@
...
+    youtubeId: selectors.app.blockValue.data.metadata.youtube_id_1_0,
+    html5Sources: selectors.app.blockValue.data.metadata.html5_sources,
+  });
+
+  // we don't appear to want to parse license version
+  const [licenseType, licenseOptions] = module.parseLicense(selectors.app.blockValue.data.metadata.license);
+
+  dispatch(actions.video.load({
+    videoSource,
+    fallbackVideos,
```
Comment (KristinAoki) [inline src/editors/data/redux/thunkActions/video.js:20]: Do we need to add videoId here?

Comment (rayzhou-bit) [reply src/editors/data/redux/thunkActions/video.js:22]: Good catch. I put it in

**Review by KristinAoki — CHANGES_REQUESTED** [review]

File: src/editors/data/redux/thunkActions/video.js [hunk 2]
```diff
@@ -1,9 +1,131 @@
...
+import * as module from './video';
 
 export const loadVideoData = () => (dispatch) => {
-  dispatch(actions.video.load(singleVideoData));
+  const {
+    videoSource,
+    videoId,
+    fallbackVideos,
+  } = module.determineVideoSource({
+    edxVideoId: selectors.app.blockValue.data.metadata.edx_video_id,
```
[the remark this answers is not in the document]
Comment (rayzhou-bit) [reply src/editors/data/redux/thunkActions/video.js]: I updated based on how you coded `deleteTranscript()` with `getState`. Updated tests and mockApi as well.

**Review by KristinAoki — APPROVED** [review]

**Pull request merged**